asa_parser/src/asa_parse.py

```diff
@@ -200,9 +200,55 @@ def show_memory_region(self):
                 memory_region.append(mem)
         return json.dumps(memory_region)
 
+    @property
     def show_cpu_detailed(self):
-        """Parser for show cpu detailed"""
-        return json.dumps({'text': self.get_show_section('cpu detailed')})
+        """
+        Parser for show cpu detailed
+        The sections of show cpu detailed are parsed according to the break down of per core data path,
+        current control point elapsed,cpu utilization for external processes, and total cpu utilization
+        """
+        cpu_detailed = []
+        #Dictionary to hold core data
+        cores_data = {}
+
+        breakdown = {}
+        total_cpu_utilization = {}
+        external_cpu_utilization = {}
+        current_control_point = {}
+
+        prev_line = ""
+        temp_string = ""
+        present = False
+        for line in self.get_show_section("cpu detailed"):
+            x = re.search("^Core\s?\d+",line)
+            present = bool(x)
+            if(present):
+                #Use the CORE and Number as the key for the dictionary
+                #The dictionary contains information about the core for 5 sec,1 min and 5 min
+                key = re.match("^Core\s+\d+",line).group()
+                attributes = line.split(key)
+                cores_data[key] = re.sub("\s+"," ",attributes[1])
+            if prev_line.startswith("Break"):
+                temp_string = re.sub("\s+"," ",line)
+                temp_string = prev_line + temp_string
```

Comment:
I think you meant to set breakdown here

Reply:
I meant to put the breakdown section together at the very end because I knew that I would still have to process core data. But I have revised it now so that once I move onto the second section that I am parsing,(the section beginning with Current control point elapsed versus the maximum control point elapsed for:). I think this is better than what I had previously ,but am still unsure if it is as efficient as it could be.